### PR TITLE
Backport v0.33: pebble chunk data packs

### PR DESCRIPTION
**Pull request by zhangchiqing**
Backport https://github.com/onflow/flow-go/pull/6042 to v0.33

In addition, in order to be backward compatible, if the folder has been used by badger database, then continue using badger, otherwise use pebble database for chunk data packs.

### CODE REVIEW

File: cmd/execution_builder.go [hunk 1]
```diff
@@ -699,17 +724,51 @@ func (exeNode *ExecutionNode) LoadExecutionState(
...
+	var closer io.Closer
+	var chunkDataPacks storageerr.ChunkDataPacks
+	node.Logger.Info().
+		Bool("is_pebble", isPebble).
+		Bool("is_badger", isBadger).
+		Bool("is_empty", isEmpty).
+		Str("chunk_data_pack_dir", exeNode.exeConf.chunkDataPackDir).
+		Msgf("opening chunk data pack database")
+
+	if isBadger {
```
Comment (zhangchiqing) [inline cmd/execution_builder.go:748]: This part is the new change that only for v0.33 to be backward compatible, which [this PR does not have](https://github.com/onflow/flow-go/pull/6042)

File: storage/folder.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	for _, file := range files {
+		fileName := file.Name()
+		switch {
+		case strings.HasPrefix(file.Name(), "MANIFEST-"):
+			hasPebbleManifest = true
+		case fileName == "MANIFEST":
+			hasBadgerManifest = true
+		case fileName == "CURRENT":
+			hasCurrent = true
+		case file.Name() == "KEYREGISTRY":
```
Comment (peterargue) [inline storage/folder.go:55]:
```suggestion
		case fileName == "KEYREGISTRY":
```

File: storage/folder.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	if hasPebbleManifest && hasCurrent && hasLog {
+		isPebbleFolder = true
+	}
+
+	// Determine if the folder contains Badger data
+	if hasBadgerManifest && hasKeyRegistery && hasVlog {
+		isBadgerFolder = true
+	}
+
+	return isPebbleFolder, isBadgerFolder, false, nil
```
Comment (peterargue) [inline storage/folder.go:74]: it's possible that the folder has both badger and pebble files. Should that be supported?

File: storage/folder_test.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+		require.False(t, isEmpty)
+		require.True(t, isBadger)
+		require.False(t, isPebble)
+	})
+}
+
+func TestCheckFolderIsPebble(t *testing.T) {
+	unittest.RunWithTempDir(t, func(dir string) {
+		fmt.Println("dir", dir)
```
Comment (peterargue) [inline storage/folder_test.go:41]:
```suggestion
```

File: utils/unittest/unittest.go [hunk 5]
```diff
@@ -314,7 +314,7 @@ func TempDir(t testing.TB) string {
 func RunWithTempDir(t testing.TB, f func(string)) {
 	dbDir := TempDir(t)
 	defer func() {
-		require.NoError(t, os.RemoveAll(dbDir))
+		// require.NoError(t, os.RemoveAll(dbDir))
```
Comment (peterargue) [inline utils/unittest/unittest.go:317]:
```suggestion
		require.NoError(t, os.RemoveAll(dbDir))
```

**Review by peterargue — APPROVED** [review]
added a couple small comments. my main question is about allowing both pebble and badger at the same time.

File: storage/folder.go (hunk 3 above)
Comment (zhangchiqing) [reply storage/folder.go:74]: Good point. We should not support that. I will add a check that stops when both are true

**Review by ramtinms — APPROVED** [review]

**Pull request merged**